src/test/scala/nl.knaw.dans.easy.dd2d/DdmToDataverseMapperSpec.scala

```diff
@@ -124,4 +126,141 @@ class DdmToDataverseMapperSpec extends TestSupportFixture with BlockCitation wit
+                      <dcx-dai:role xml:lang="en">Related Person</dcx-dai:role>
+                  </dcx-dai:author>
+              </dcx-dai:contributorDetails>
+              <dcx-dai:contributorDetails>
+                  <dcx-dai:organization>
+                      <dcx-dai:name xml:lang="en">Advocatenkantoor Harker</dcx-dai:name>
+                      <dcx-dai:role xml:lang="en">RightsHolder</dcx-dai:role>
+                  </dcx-dai:organization>
+              </dcx-dai:contributorDetails>
+              <dcterms:hasVersion scheme="DOI">https://doi.org/10.17632/5x6xt4zhws.3</dcterms:hasVersion>
+              <dcterms:references href="https://a-v.l.w.com">Anti-Vampire League website</dcterms:references>
+              <dcterms:dateAccepted>2016-09-08</dcterms:dateAccepted>
+              <dcterms:modified xsi:type="W3CDTF">2017-01-02</dcterms:modified>
+              <dcterms:identifier xsi:type="ARCHIS-ZAAK-IDENTIFICATIE">1234567890</dcterms:identifier>
+              <dcx-gml:spatial>
+                  <Point xmlns="http://www.opengis.net/gml">
+                      <pos>
+                          52.08113 4.34510
+                      </pos>
+                  </Point>
+              </dcx-gml:spatial>
+                  <dcx-gml:spatial>
+                      <boundedBy xmlns="http://www.opengis.net/gml">
+                          <Envelope srsName="urn:ogc:def:crs:EPSG::28992">
+                              <lowerCorner>91232.015554 436172.485680</lowerCorner>
+                              <upperCorner>121811.885272 486890.494251</upperCorner>
+                          </Envelope>
+                      </boundedBy>
+                  </dcx-gml:spatial>
+          </ddm:dcmiMetadata>
+      </ddm:DDM>
+
+    val metadataBlocks = mapper.toDataverseDataset(ddm).get.datasetVersion.metadataBlocks
+    val citationJson = Extraction.decompose(metadataBlocks("citation"))
+    debug(Serialization.writePretty(citationJson))
+
+    (citationJson \ "displayName") shouldBe JString("Citation Metadata")
+
+    /*
+     * Expected:
+     * {
+     *    "typeName":"title",
+     *    "multiple":false,
+     *    "typeClass":"primitive",
+     *    "value":"A title"
+     * }
+     */
+    val titles = (citationJson \ "fields").filter(_ \ "typeName" == JString("title"))
+    titles should have length (1)
+    val title = titles.head
+    title \ "multiple" shouldBe JBool(false)
+    title \ "typeClass" shouldBe JString("primitive")
+    title \ "value" shouldBe JString("A title")
+
+    /*
+     * Expected:
+     * {
+     *    "typeName":"otherId",
+     *    "multiple":true,
+     *    "typeClass":"compound",
+     *    "value":[
+     *        {
+     *          "otherIdValue":{
+     *              "typeName":"otherIdValue",
+     *              "multiple":false,
+     *              "typeClass":"primitive",
+     *              "value":"https://test.example/1"
+     *           }
+     *        }
+     *     ]
+     * }
+     */
```

Comment:
Wild idea: can't you serialize `citationJson`, split into lines and test whether it contains all lines of the expected result? Much easier to read and allows for random order of lines 242-245  respective 248-251.